core/trino-main/src/main/java/io/trino/server/protocol/spooling/encoding/JsonQueryDataEncoder.java

```diff
@@ -127,7 +133,7 @@ private static void writeValue(JsonGenerator generator, Object value)
                 }
                 generator.writeEndObject();
             }
-            default -> throw new UnsupportedOperationException("Unsupported type: " + value.getClass().getName());
+            default -> generator.writeString(value.toString());
```

Comment:
This needs to use `ObjectMapper` for types like `BingTile` that use custom serialization.

Reply:
Done. Makes sense.